### ISSUE TITLE
Arrays in return value of GM_getValue do not satisfy instanceof Array

### ISSUE THREAD

**Issue by ttsilva90** [Tampermonkey/tampermonkey#1409]
This issue first started occurring today when I updated both Chrome and Tampermonkey to the latest stable versions (described below). An example of an issue that it causes is when using the [mapping library for Knockout.js](https://knockoutjs.com/documentation/plugins-mapping.html), which checks `instanceof Array` when executing the "create" callback.

## Expected Behavior
`instanceof Array` returns `true` on an object returned by GM_getValue if it is an array

## Actual Behavior
`instanceof Array` returns `false` on an object returned by GM_getValue even though it is an array

## Specifications

- Chrome: 98.0.4758.80
- TM: 4.1.4
- OS: macOS 11.6.2

## Script

```js
// ==UserScript==
// @name         Test GM_getValue Array
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       You
// @match        *://github.com/*
// @grant        GM_getValue
// @grant        GM_setValue
// ==/UserScript==

(function () {
    'use strict';

    GM_setValue('foo', ['bar']);
    console.assert(GM_getValue('foo') instanceof Array); // Fails

    GM_setValue('test', { hello: ['world'] });
    console.assert(GM_getValue('test').hello instanceof Array); // Fails
})();
```

**Comment by derjanb** [comment]
As a workaround: please add or `@require` [something like this](https://gist.githubusercontent.com/derjanb/1d490a4a0986aa3fd429968c9375899d/raw/fix_gm_getvalue_array_types.js) in your script. 

```JS
// ==UserScript==
// @name         Test GM_getValue Array
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       You
// @match        *://github.com/*
// @grant        GM_getValue
// @grant        GM_setValue
// @require      https://gist.githubusercontent.com/derjanb/1d490a4a0986aa3fd429968c9375899d/raw/fix_gm_getvalue_array_types.js
// ==/UserScript==

(function () {
    'use strict';

    GM_setValue('foo', ['bar']);
    console.assert(GM_getValue('foo') instanceof Array); // Fails

    GM_setValue('test', { hello: ['world'] });
    console.assert(GM_getValue('test').hello instanceof Array); // Fails
})();
```

**Comment by BogudanPro** [comment]
Bug confirmed for Chrome 98, TM 4.14, Windows 7. Fix solves issue.

There is also `GM_getValue ('test') instanceof Object` bug: returns `false` without fix, returns `true` with fix applied.

**Comment by derjanb** [comment]
Should be fixed at 4.15.6153 (in review|[crx](https://tampermonkey.net/crx/tampermonkey_beta_4_15_6153.crx))
